From 4c7320addc06e608a2ce633cb3356f77eb52399a Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Wed, 1 Jul 2026 11:18:53 +0200
Subject: [PATCH] feat(stm32mp2): do not disable PM with STM32MP_USB_PROGRAMMER
 enabled

Keep PM functionality enabled when STM32MP_USB_PROGRAMMER is set.
This is a convenience patch which allows build of evaluation boards
with support for boot from storage and USB DFU.

This change may have side effects, do not use in production.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 plat/st/common/common.mk            | 16 ++++++++--------
 plat/st/stm32mp2/bl2_plat_setup.c   |  6 +++---
 plat/st/stm32mp2/bl31_plat_setup.c  |  4 ++--
 plat/st/stm32mp2/platform.mk        |  8 ++++----
 plat/st/stm32mp2/stm32mp2_private.c |  2 +-
 5 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/plat/st/common/common.mk b/plat/st/common/common.mk
index a2e9272840..e61c788247 100644
--- a/plat/st/common/common.mk
+++ b/plat/st/common/common.mk
@@ -52,12 +52,12 @@ STM32MP_EMMC_BOOT		?=	0
 STM32MP_UART_PROGRAMMER		?=	0
 STM32MP_USB_PROGRAMMER		?=	0
 
-ifneq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
-ifeq (${PSA_FWU_SUPPORT},1)
-$(info Disable PSA_FWU_SUPPORT flag for serial device)
-override PSA_FWU_SUPPORT	:=	0
-endif
-endif
+# ifneq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
+# ifeq (${PSA_FWU_SUPPORT},1)
+# $(info Disable PSA_FWU_SUPPORT flag for serial device)
+# override PSA_FWU_SUPPORT	:=	0
+# endif
+# endif
 
 # Number of TF-A copies in the device
 STM32_TF_A_COPIES		:=	2
@@ -287,9 +287,9 @@ endif
 BL2_SOURCES			+=	drivers/mtd/hyperflash/hyperflash.c
 endif
 
-ifneq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
+# ifneq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
 BL2_SOURCES			+=	drivers/io/io_memmap.c
-endif
+# endif
 
 ifeq (${STM32MP_UART_PROGRAMMER},1)
 BL2_SOURCES			+=	plat/st/common/stm32cubeprogrammer_uart.c
diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index efdfb312b0..c9c07a8376 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -621,7 +621,7 @@ skip_console_init:
 	}
 #endif
 
-#if STM32MP_DDR_FIP_IO_STORAGE && !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER
+#if STM32MP_DDR_FIP_IO_STORAGE /* && !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER */
 	/* Skip DDR FW ID = the first image to load for standby exit */
 	if (stm32mp_is_wakeup_from_standby()) {
 		bl_mem_params_node_t *bl_mem_params;
@@ -640,7 +640,7 @@ static void prepare_encryption(void)
 #if !STM32MP_M33_TDCID
 	uint8_t mkey[RISAF_KEY_SIZE_IN_BYTES];
 
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
+#if 0 /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 	if (stm32_rng_read(mkey, RISAF_KEY_SIZE_IN_BYTES) != 0) {
 		panic();
 	}
@@ -727,7 +727,7 @@ int bl2_plat_handle_post_image_load(unsigned int image_id)
 				FW_CONFIG_ID);
 		fconf_populate("FW_CONFIG", STM32MP_FW_CONFIG_BASE);
 
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
+#if 0 /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 		wakeup_standby = false;
 #else /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 		wakeup_standby = stm32mp_is_wakeup_from_standby();
diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index ecf6ba6d93..ae574d3fa8 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -150,7 +150,7 @@ void bl31_plat_arch_setup(void)
 
 	stm32mp_gic_init();
 
-#if !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER
+#if 1 /* !STM32MP_UART_PROGRAMMER && !STM32MP_USB_PROGRAMMER */
 	if (stm32mp_is_wakeup_from_standby()) {
 		/* Initialize the runtime services e.g. PSCI. */
 		runtime_svc_init();
@@ -213,7 +213,7 @@ entry_point_info_t *bl31_plat_get_next_image_ep_info(unsigned int type)
 }
 
 /* Only support system reset for serial boot and no low power feature to reduce BL31 size */
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
+#if 0 /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 static void __dead2 stm32_system_reset(void)
 {
 	stm32mp_system_reset();
diff --git a/plat/st/stm32mp2/platform.mk b/plat/st/stm32mp2/platform.mk
index ce1974f013..0496cdba55 100644
--- a/plat/st/stm32mp2/platform.mk
+++ b/plat/st/stm32mp2/platform.mk
@@ -315,9 +315,9 @@ PLAT_BL_COMMON_SOURCES		+=	drivers/st/clk/clk-stm32-core.c				\
 
 PLAT_BL_COMMON_SOURCES  	+=	drivers/st/nvmem/stm32mp_tamp_nvram_mp2.c
 
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
+# ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
 PLAT_BL_COMMON_SOURCES  	+=	plat/st/stm32mp2/stm32mp2_context.c
-endif
+# endif
 
 BL2_SOURCES			+=	plat/st/stm32mp2/plat_bl2_mem_params_desc.c
 
@@ -412,13 +412,13 @@ BL31_SOURCES			+=	${FDT_WRAPPERS_SOURCES}
 BL31_SOURCES			+=	plat/st/stm32mp2/bl31_plat_setup.c			\
 					plat/st/stm32mp2/stm32mp2_topology.c
 
-ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
+# ifeq ($(filter 1,${STM32MP_UART_PROGRAMMER} ${STM32MP_USB_PROGRAMMER}),)
 BL31_SOURCES			+=	plat/st/stm32mp2/stm32mp2_pm.c				\
 					plat/st/stm32mp2/stm32mp2_ca35ss.c			\
 					common/tf_crc32.c
 
 BL31_CPPFLAGS += -march=armv8-a+crc
-endif
+# endif
 
 ifeq ($(STM32MP_M33_TDCID),0)
 BL31_SOURCES			+=	drivers/st/ddr/stm32mp2_ddr_helpers.c
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index f1c7d53d4c..8372857f6f 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -655,7 +655,7 @@ bool stm32mp_skip_boot_device_after_standby(void)
 
 bool stm32mp_is_wakeup_from_standby(void)
 {
-#if STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER
+#if 0 && (STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER)
 	return false;
 #else /* STM32MP_UART_PROGRAMMER || STM32MP_USB_PROGRAMMER */
 	static int standby = -1;
-- 
2.53.0

